crates/nu-cli/src/completion.rs

```diff
@@ -0,0 +1,26 @@
+use nu_errors::ShellError;
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct Suggestion {
+    pub display: String,
+    pub replacement: String,
+}
+
+pub struct Context<'a>(pub &'a rustyline::Context<'a>);
+
+impl<'a> AsRef<rustyline::Context<'a>> for Context<'a> {
+    fn as_ref(&self) -> &rustyline::Context<'a> {
+        self.0
+    }
+}
```

Comment:
Making the context be just a wrapper around rustyline's context makes life a lot simpler for now. This will likely change once we have our own completer.